### PR TITLE
Bug fixes3

### PR DESCRIPTION
**Pull request by au1994**
@vaidik
@vishesh92

### CODE REVIEW

File: movie.sh [hunk 1]
```diff
@@ -1,6 +1,53 @@
...
+declare -r ZERO="0"; 
+
+
+#Print the heading first
+#printf "%-10s  %-30s  %-30s\n" "RATING" "MOVIE" "TITLE"
+printline "RATING" "MOVIE" "TITLE";
+
+for f in $dir/*
+  do
+     #strip initial path value to get the movie name
```
Comment (vaidik) [inline movie.sh]: Line 33 to line 49 has 5 spaces. It should be 4 spaces. Fix this please.

File: movie.sh [hunk 2]
```diff
@@ -1,6 +1,53 @@
...
+     search=$(echo $movie_name | tr ._' ' +)
+     
+     #Curl the api to get the json data
+     rating=$(curl -s http://www.omdbapi.com/\?i\=\&t\=$search | jq -r '.imdbRating');
+     title=$(curl -s http://www.omdbapi.com/\?i\=\&t\=$search | jq -r '.Title');
+
+     #check if rating is null (if curl fails)
+     if [ "$rating" == "$NULL" ] || [ "$rating" == "$EMPTY" ] || [ "$rating" == "$ZERO" ]
+     then
+	printline "$NA" "$movie_name" "$title";
```
Comment (vaidik) [inline movie.sh]: This line should have 6 spaces.

File: movie.sh [hunk 3]
```diff
@@ -1,6 +1,53 @@
...
+     #Curl the api to get the json data
+     rating=$(curl -s http://www.omdbapi.com/\?i\=\&t\=$search | jq -r '.imdbRating');
+     title=$(curl -s http://www.omdbapi.com/\?i\=\&t\=$search | jq -r '.Title');
+
+     #check if rating is null (if curl fails)
+     if [ "$rating" == "$NULL" ] || [ "$rating" == "$EMPTY" ] || [ "$rating" == "$ZERO" ]
+     then
+	printline "$NA" "$movie_name" "$title"; 
+     else 
+	printline "$rating" "$movie_name" "$title"
```
Comment (vaidik) [inline movie.sh]: This line as well should have 6 spaces.

File: movie.sh [hunk 4]
```diff
@@ -1,6 +1,53 @@
 #!/bin/bash
-echo 'please input the movie name'
-read name
-echo $name
-var=$(curl -s http://www.omdbapi.com/\?i\=\&t\=$name | jq -r '.imdbRating')
-echo $var
+function printline()
+{
+   printf "%-10s  %-30s  %-30s\n" "$1" "$2" "$3";
+}
```
Comment (vaidik) [inline movie.sh:5]: One blank line after the function ends

File: movie.sh [hunk 5]
```diff
@@ -1,6 +1,52 @@
...
+for f in $dir/*
+do
+    #strip initial path value to get the movie name
+    movie_name=${f##*/}
+
+    #Replace any dot, underscore or space with '+'
+    search=$(echo $movie_name | tr ._' ' +)
+     
+    #Curl the api to get the json data
+    rating=$(curl -s http://www.omdbapi.com/\?i\=\&t\=$search | jq -r '.imdbRating');
```
Comment (vishesh92) [inline movie.sh]: You are making 2 api calls here. This will slow down your application.

File: movie.sh [hunk 6]
```diff
@@ -1,6 +1,52 @@
...
+    title=$(curl -s http://www.omdbapi.com/\?i\=\&t\=$search | jq -r '.Title');
+
+    #check if rating is null (if curl fails)
+    if [ "$rating" == "$NULL" ] || [ "$rating" == "$EMPTY" ] || [ "$rating" == "$ZERO" ]
```
Comment (vishesh92) [inline movie.sh:52]: Line length

**Pull request merged**